### PR TITLE
Feat: DO-3135: Add powered by causaLens logo

### PR DESCRIPTION
**Pull request by Roman-Kornev**
## Motivation and Context

- Adds a "Powered by causaLens" logo in the sidebar to apps when `powered_by_causalens` flag in config is enabled. To use it in your apps set `config.powered_by_causalens = True`.
- Adds a link to our github to "Built with Dara" logo.

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [x] I have added relevant tests (unit, integration or regression).
- [x] I have added comments to all the bits that are hard to follow.
- [x] I have added/updated Documentation.
- [x] I have updated the appropriate changelog with a line for my changes.

## Screenshots (if appropriate):

image
image
Darkmode:
image

### CODE REVIEW

File: packages/dara-core/js/components/side-bar-frame/side-bar-frame.tsx [hunk 1]
```diff
@@ -54,7 +56,7 @@ const SideBar = styled.div.withConfig({ shouldForwardProp })<SideBarProps>`
 
 const LogoutButton = styled(Button)`
     width: 80%;
-    margin-bottom: 1rem;
+    margin-bottom: 1.5rem;
```
[the remark this answers is not in the document]
Comment (Roman-Kornev) [reply packages/dara-core/js/components/side-bar-frame/side-bar-frame.tsx:59]: This was designer feedback. Here's how it looks:
image
image

File: packages/dara-core/js/components/side-bar-frame/side-bar-frame.tsx [hunk 2]
```diff
@@ -70,11 +72,17 @@ const LogoutButton = styled(Button)`
...
+const BuiltWithLink = styled.a`
     display: flex;
     gap: 0.2rem;
     align-items: center;
     font-size: 0.75rem;
+    text-decoration: none;
+    color: inherit;
+
+    :hover {
+        text-decoration: underline;
```
[the remark this answers is not in the document]
Comment (Roman-Kornev) [reply packages/dara-core/js/components/side-bar-frame/side-bar-frame.tsx:84]: No, as I override it 2 lines above

[the remark this answers is not in the document]
Comment (krzysztof-causalens) [reply packages/dara-core/js/components/side-bar-frame/side-bar-frame.tsx:84]: Right, missed that somehow

File: packages/dara-core/js/components/side-bar-frame/side-bar-frame.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (krzysztof-causalens) [reply packages/dara-core/js/components/side-bar-frame/side-bar-frame.tsx:59]: Fair enough, looks good

**Review by tamara-causalens — APPROVED** [review]
Very nice! 😍

**Pull request merged**